### PR TITLE
Feature/v3.11.x/1115 osx executable args issue

### PR DESCRIPTION
**Pull request by witcher112**
### Checklist
1. [x] PR associated with Redmine issue (if applicable)
2. [x] Changelog entry with id of Redmine issue
3. [x] `Version.cs` up to date
4. [x] Review requested

### CODE REVIEW

**Comment by witcher112** [comment]
No changelog update since it's a fix for feature that is currently being added.

File: Assets/PatchKit Patcher/Scripts/AppStarter.cs [hunk 1]
```diff
@@ -29,11 +29,17 @@ public AppStarter(App app)
...
         private string ResolveExecutablePath(AppVersion appVersion)
         {
+            PlatformType platformType = Platform.GetPlatformType();
+
             if (!string.IsNullOrEmpty(appVersion.MainExecutable))
             {
                 string executablePath = Path.Combine(_app.LocalDirectory.Path, appVersion.MainExecutable);
 
-                if (File.Exists(executablePath))
+                bool isOSXApp = platformType == PlatformType.OSX &&
```
Comment (Dzejkop) [inline Assets/PatchKit Patcher/Scripts/AppStarter.cs]: Is there any chance that `Platform.GetPlatformType` returns `OSX` but the application isn't an OSX app?

Comment (witcher112) [reply Assets/PatchKit Patcher/Scripts/AppStarter.cs]: Nope, it's always the same.

Comment (Dzejkop) [reply Assets/PatchKit Patcher/Scripts/AppStarter.cs]: The we shouldn't need all the other conditions, right?

 ```
...
executablePath.EndsWith(".app") &&
Directory.Exists(executablePath);
```

Comment (witcher112) [reply Assets/PatchKit Patcher/Scripts/AppStarter.cs]: This isn't checking whether application platform that secret we are using is OSX, but whether the `appVersion.MainExecutable` points to valid OSX `.app`.

**Pull request merged**